pkg/armhelpers/utils/util.go

```diff
@@ -160,3 +160,26 @@ func GetK8sVMName(p *api.Properties, agentPoolProfile *api.AgentPoolProfile, age
 
 	return "", errors.Errorf("Failed to reconstruct VM Name")
 }
+
+// IsVMSSUpgradeable - checks if the current vm(in case of VMA) or vm instance (in case of VMSS)
+// is specified in the upgradeMap.
+func IsVMSSUpgradeable(upgradeMap map[string]bool, vmssName string) bool {
```

Comment:
I would rename this function to something other than upgradable - upgradable sounds like it's checking if it _can_ be upgraded as opposed to what it's actually doing: checking if it _should_ be upgraded. The name also conflicts with pre-existing [upgradable](https://github.com/Azure/aks-engine/blob/master/pkg/operations/kubernetesupgrade/upgradecluster.go#L215) function which actually does check if a node can be upgraded.